test/integration/02-user-wallet/02-index.spec.ts

```diff
@@ -0,0 +1,177 @@
+
+    expect(userWallet2.user.oldEnoughForWithdrawal).toBeFalsy()
+  })
+
+  it("allows withdraw if user is old enough", async () => {
+    expect(userWallet2.user.oldEnoughForWithdrawal).toBeFalsy()
+
+    // TODO make this configurable
+    // in 8 days:
+    const date = Date.now() + yamlConfig.limits.oldEnoughForWithdrawal + 60 * 60 * 1000
+
+    jest.spyOn(global.Date, "now").mockImplementationOnce(() => new Date(date).valueOf())
+
+    expect(userWallet2.user.oldEnoughForWithdrawal).toBeTruthy()
+  })
+
+  describe("setUsername", () => {
+    it("does not set username if length is less than 3", async () => {
+      await expect(userWallet0.setUsername({ username: "ab" })).rejects.toThrow()
+    })
+
+    it("does not set username if contains invalid characters", async () => {
+      await expect(userWallet0.setUsername({ username: "ab+/" })).rejects.toThrow()
+    })
+
+    it("does not allow non english characters", async () => {
+      await expect(userWallet0.setUsername({ username: "ñ_user1" })).rejects.toThrow()
+    })
+
+    it("does not set username starting with 1, 3, bc1, lnbc1", async () => {
+      await expect(userWallet0.setUsername({ username: "1ab" })).rejects.toThrow()
+      await expect(userWallet0.setUsername({ username: "3basd" })).rejects.toThrow()
+      await expect(userWallet0.setUsername({ username: "bc1ba" })).rejects.toThrow()
+      await expect(userWallet0.setUsername({ username: "lnbc1qwe1" })).rejects.toThrow()
+    })
+
+    it("allows set username", async () => {
+      let result = await userWallet0.setUsername({ username: "user0" })
+      expect(!!result).toBeTruthy()
+      result = await userWallet1.setUsername({ username: "user1" })
+      expect(!!result).toBeTruthy()
+      result = await userWallet2.setUsername({ username: "lily" })
+      expect(!!result).toBeTruthy()
+    })
+
+    it("does not allow set username if already taken", async () => {
+      await getUserWallet(2)
+      await expect(userWallet2.setUsername({ username })).rejects.toThrow()
+    })
+
+    it("does not allow set username with only case difference", async () => {
+      await expect(userWallet2.setUsername({ username: "User1" })).rejects.toThrow()
+    })
+
+    it("does not allow re-setting username", async () => {
+      await expect(userWallet0.setUsername({ username: "abc" })).rejects.toThrow()
+    })
+  })
+
+  describe("usernameExists", () => {
+    it("return true if username already exists", async () => {
+      const result = await usernameExists({ username })
+      expect(result).toBe(true)
+    })
+
+    it("return true for other capitalization", async () => {
+      const result = await usernameExists({ username: username.toLocaleUpperCase() })
+      expect(result).toBe(true)
+    })
+
+    it("return false if username does not exist", async () => {
+      const result = await usernameExists({ username: "user" })
+      expect(result).toBe(false)
+    })
+  })
+
+  describe("getOnchainFee", () => {
```

Comment:
I would feel the getOnchainFee, getInvoice would benefits to be independant/in another file of the username stuff. one is purely mongodb related, the other involved bitcoind and lnd. what do you think?

Reply:
Fixed https://github.com/GaloyMoney/galoy/pull/340/commits/e27cfbf08c2c0eb3d39671d82592efd6fec82653